src/Model/ViewModel.php

```diff
@@ -239,8 +239,13 @@ public function clearOptions()
     public function getVariable($name, $default = null)
     {
         $name = (string) $name;
-        if (isset($this->variables[$name])) {
-            return $this->variables[$name];
+
+        if (is_array($this->variables)) {
+            if (array_key_exists($name, $this->variables)) {
+                return $this->variables[$name];
+            }
+        } elseif ($this->variables->offsetExists($name)) {
```

Comment:
Don't you want to check if this is an `ArrayAccess` object first ?

Reply:
As I can see from the code (construct, setter...) it must be an array or `Zend\View\Variables` (extends `ArrayObject`), so I think no additional check is needed here.